eng/linkerTests/linkerCorrectnessTests.targets

```diff
@@ -0,0 +1,113 @@
+    <MakeDir Directories="%(TestSupportFiles.DestinationFolder)" />
+    <Copy SourceFiles="@(TestSupportFiles)" DestinationFolder="%(TestSupportFiles.DestinationFolder)" />
+  </Target>
+
+  <Target Name="GetTestConsoleApps">
+    <ItemGroup>
+      <TestConsoleAppSourceFiles>
+        <ProjectDir>$(LinkerTestProjectsDir)$(MSBuildProjectName)/%(Filename)/</ProjectDir>
+      </TestConsoleAppSourceFiles>
+      <TestConsoleAppSourceFiles>
+        <ProjectFile>%(ProjectDir)project.csproj</ProjectFile>
+        <AssetsFile>%(ProjectDir)obj/project.assets.json</AssetsFile>
+        <TestCommand>%(ProjectDir)bin\Release\net5.0\win-x64\publish\CoreRun.exe /v project.dll</TestCommand>
+        <TestExecutionDirectory>%(ProjectDir)bin\Release\net5.0\win-x64\publish\</TestExecutionDirectory>
+      </TestConsoleAppSourceFiles>
+    </ItemGroup>
+
+    <ItemGroup>
+      <TestConsoleApps Include="@(TestConsoleAppSourceFiles->'%(ProjectFile)')">
+        <ProjectCompileItems>%(Identity)</ProjectCompileItems>
+      </TestConsoleApps>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="GenerateProjects"
+          DependsOnTargets="GetTestConsoleApps;CreateTestDir"
+          Inputs="@(TestConsoleAppSourceFiles);$(ProjectTemplate)"
+          Outputs="%(TestConsoleApps.Identity)">
+    <PropertyGroup>
+      <_projectDir>%(TestConsoleApps.ProjectDir)/</_projectDir>
+      <_projectFile>%(TestConsoleApps.ProjectFile)</_projectFile>
+      <_projectSourceFile>%(TestConsoleApps.ProjectCompileItems)</_projectSourceFile>
+    </PropertyGroup>
+
+    <MakeDir Directories="$(_projectDir)" />
+    <WriteLinesToFile File="$(_projectFile)"
+                      Lines="$([System.IO.File]::ReadAllText('$(ProjectTemplate)').Replace('{RuntimePackDir}', '$(RuntimePackDir)'))"
+                      Overwrite="true" />
+    <Copy SourceFiles="$(_projectSourceFile)"
+          DestinationFolder="$(_projectDir)" />
+    <Message Text="Generated $(_projectFile)" />
+  </Target>
+
+  <Target Name="RestoreProjects"
+          DependsOnTargets="GenerateProjects">
+    <PropertyGroup>
+      <TestRestoreCommand>"$(TestDotNetPath)"</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) restore</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) /nr:false</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) /warnaserror</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) -p:configuration=Release</TestRestoreCommand>
+    </PropertyGroup>
+
+    <Exec Command="$(TestRestoreCommand)" StandardOutputImportance="Low" WorkingDirectory="%(TestConsoleApps.ProjectDir)" />
+  </Target>
+
+  <Target Name="PublishTrimmedProjects"
+          DependsOnTargets="RestoreProjects">
+    <PropertyGroup>
+      <TestRestoreCommand>"$(TestDotNetPath)"</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) msbuild /t:PublishTrimmed</TestRestoreCommand>
```

Comment:
Why is this `/t:PublishTrimmed` and not `/t:Publish /p:PublishTrimmed=true`

Reply:
This is because we need to do some extra manual override of the shared framework that will be used for publishing as we want to force the use of the live-built shared framework as opposed to what gets resolved by the SDK. These are the lines that are doing that before publishing: https://github.com/dotnet/runtime/pull/37618/files#diff-63703354eec4d5605432a9a6ed456138R2-R15